File: MatasanoCrypto/matasano7.py
import binascii, base64
from Crypto.Cipher import AES

# Using Python 2.7 because I could only install PyCrypto on 2.7
# Matasano Crypto Challenges
# Challenge 7 - AES in ECB mode

def aesDecryptECB(ciphertext, key):
    """
    ciphertext = string to be decrypted
    key = string key, 16 bytes long plz
    """
    cipher = AES.new(key, AES.MODE_ECB)
    return cipher.decrypt(ciphertext)

# --- Main / Decrypt ----

theKey = "YELLOW SUBMARINE"

ciphertext = str()
with open("7.txt") as f:
    for line in f:
        ciphertext += base64.b64decode(line.strip())

# print(ciphertext)
try:
    print(aesDecryptECB(ciphertext, theKey))
except ValueError:
    print("Value Error, likely odd length string")

File: MatasanoCrypto/matasano7.js
/**
 * Matasano Crypto Challenges
 * Challenge 7 - AES in ECB mode
 * 
 * This is currently NOT working correctly! :(
 */
"use strict";
const crypto = require('crypto');
const assert = require('assert');
const fs = require('fs');

// Start main part / Decryption
const theKey = "YELLOW SUBMARINE";

let ciphertextBuf = Buffer.from(fs.readFileSync('inputData/7.txt'), 'base64');

console.log(decryptAesEcb(ciphertextBuf, theKey));

/**
 * ciphertext - string to be decrypted.
 * key - string key, 16 bytes long.
 */
function decryptAesEcb(ciphertextBuffer, key) {
  const algorithm = 'aes-128-ecb';
  let decipher = crypto.createDecipher(algorithm, key);
  decipher.setAutoPadding(false);

  let deciphered = Buffer.concat([decipher.update(ciphertextBuffer), decipher.final()]);

  return deciphered.toString('base64');
}
